src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/cdk/cdk_constructs/adf_codebuild.py

```diff
@@ -373,6 +373,20 @@ def determine_build_image(codebuild_id, scope, target, map_params):
                 .get('image')
             )
         if isinstance(specific_image, dict):
+            response_specific_image = specific_image['repository_arn']
+            # We just use the ecr repo name and construct the ECR ARN
+            if not response_specific_image.startswith('arn:aws:ecr:'):
+                constructed_repository_arn = f"arn:aws:ecr:{ADF_DEPLOYMENT_REGION}:{ADF_DEPLOYMENT_ACCOUNT_ID}:{response_specific_image}"
+                repo_arn = _ecr.Repository.from_repository_arn(
+                    scope,
+                    f'custom_repo_{codebuild_id}',
+                    constructed_repository_arn
+                )
+                return _codebuild.LinuxBuildImage.from_ecr_repository(
+                    repo_arn,
+                    specific_image.get('tag', 'latest'),
+                )
+            # We take the full ECR Arn - Default Behaviour
```

Comment:
Great addition, I just prefer that we change it slightly.
At the moment, we are using the `repository_arn` as a way to pass the name of the ECR repository.
The name of the property does not make this clear.

I would rather have an additional property, which is `repository_name` or something similar.
If that is specified, you build up the repository arn the way you did.

After constructing the repository arn, could you make sure we reuse the from_repository_arn and from_ecr_repository calls that were there already?

So basically, if the repository_arn property is specified we use that.
Else, if the repository_name is specified we build up the arn that way.
If both are not specified, we raise an error.